core/src/main/scala/org/apache/spark/sql/CachedDataFrame.scala

```diff
@@ -70,7 +70,13 @@ class CachedDataFrame(snappySession: SnappySession, queryExecution: QueryExecuti
   private[sql] final def isCached: Boolean = cachedRDD ne null
 
   private def getExecRDD: RDD[InternalRow] =
-    if (isCached) cachedRDD else queryExecution.executedPlan.execute()
+    if (isCached) {
+      cachedRDD
+    } else {
+      snappySession.sessionState.
+          executePlan(queryExecution.logical).executedPlan.execute()
+    }
+
```

Comment:
This bypasses most of plan caching in any case. We need to discuss and underlying issue must be fixed rather than disabling plan caching altogether. It also needs to be explored why these issues have started cropping up now and before which indicates that there has been some recent change causing these. Has someone tried to check with a git bisect?

Reply:
It still does not fail with plan caching enabled. One of the plans is the NonRecursive plan which assumes plan caching is enabled, hence it has started failing now. Same with CodeGenFallback.